src/client/src/components/homepage.js

```diff
@@ -0,0 +1,70 @@
+const React = require('react')
+const {Link, Redirect} = require('react-router-dom')
+const logo = require('../logo.svg')
+const _fetch = require('node-fetch')
+
+const LOGIN_URL = '/login', POST = 'POST', credentials = 'same-origin'
+const headers = { 'Content-Type': 'application/json' }
+
+const CHAT_PATH = '/chat', REGISTER_PATH = '/register'
+
+class HomePage extends React.Component {
+	constructor(props) {
+		super(props)
+
+		this.state = {
+			username: '',
+			password: '',
+			successfulLogin: false
+		}
+
+		this.handleChange = this.handleChange.bind(this)
+		this.handleSubmit = this.handleSubmit.bind(this)
+	}
+
+	loginHandler(username, password, fetch) {
+		fetch = fetch || /* istanbul ignore next */ _fetch
+		fetch(LOGIN_URL, {
+			method: POST, headers, credentials,
+			body: JSON.stringify({username, password})
+		}).then(response => {
+			if(response.status === 200) {
+				this.setState({successfulLogin: true})
+			}
+		}).catch(err => new Error(err))
+	}
+
+	handleChange(event) {
+		this.setState({ [event.target.name] : event.target.value})
```

Comment:
What if server will send some strange event, not predicted by client?

Reply:
I trust you enough not to send such things

Reply:
#49 will also include this